applications/FluidDynamicsApplication/custom_elements/embedded_fluid_element.h

```diff
@@ -0,0 +1,389 @@
+
+    void DefineStandardGeometryData(EmbeddedElementData& rData) const;
+
+    void DefineCutGeometryData(EmbeddedElementData& rData) const;
+
+    void NormalizeInterfaceNormals(typename EmbeddedElementData::InterfaceNormalsType& rNormals, double Tolerance) const;
+
+    /**
+    * This functions adds the penalty extra term level set contribution.
+    * @param rLHS: reference to the LHS matrix
+    * @param rRHS: reference to the RHS vector
+    * @param rData: reference to element data structure
+    */
+    void AddBoundaryConditionPenaltyContribution(
+        MatrixType& rLHS,
+        VectorType& rRHS,
+        const EmbeddedElementData& rData) const;
+    
+    /**
+     * This function computes the penalty coefficient for the level set BC imposition
+     * @param rLeftHandSideMatrix: reference to the LHS matrix
+     * @param rData: reference to element data structure
+     */
+    double ComputePenaltyCoefficient(const EmbeddedElementData& rData) const;
+
+    /**
+    * This drops the outer nodes velocity constributions in both LHS and RHS matrices.
+    * @param rLHS: reference to the LHS matrix
+    * @param rRHS: reference to the RHS vector
+    * @param rData: reference to element data structure
+    */
+    void DropOuterNodesVelocityContribution(
+        MatrixType& rLHS,
+        VectorType& rRHS,
+        const EmbeddedElementData& rData) const;
+
+    /**
+    * This functions adds the level set strong boundary condition imposition contribution.
+    * @param rLHS: reference to the LHS matrix
+    * @param rRHS: reference to the RHS vector
+    * @param rData: reference to element data structure
+    */
+    void AddBoundaryConditionModifiedNitscheContribution(
+        MatrixType& rLHS,
+        VectorType& rRHS,
+        const EmbeddedElementData& rData) const;
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ///@}
+    ///@name Serialization
+    ///@{
+
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const override;
+
+    void load(Serializer& rSerializer) override;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+
+
+    ///@}
+    ///@name Private Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    /// Assignment operator.
+    EmbeddedFluidElement& operator=(EmbeddedFluidElement const& rOther);
+
+    /// Copy constructor.
+    EmbeddedFluidElement(EmbeddedFluidElement const& rOther);
+
+    ///@}
+
+
+}; // Class EmbeddedFluidElement
+
+namespace Internals {
```

Comment:
Just for my curiosity, why you create a namespace instead of saving a member pointer inside each element? Is it because of memory consumption? Thanks!!!

Reply:
I still have a pointer inside each element (not a member, as it was not required here, but defined while calculating the local problem). This function is defined here to allow me to define an implementation that works for any TElementData but fixed geometry (that is, only for Dim=2 and NumNodes=3, for example). C++ would not allow me to define this in the element itself, because the element is templated on the complete TElementData object.

Putting it on the `Internals` namespace is done to avoid polluting the `Kratos` namespace.